language/evm/move-to-yul/src/context.rs

```diff
@@ -276,78 +276,51 @@ impl<'a> Context<'a> {
         emitln!(self.writer, "}");
     }
 
-    /// Get the field types of a struct as a vector.
-    pub fn get_field_types(&self, id: QualifiedId<StructId>) -> Vec<Type> {
-        self.env
-            .get_struct(id)
-            .get_fields()
-            .map(|f| f.get_type())
-            .collect()
-    }
-
     /// Returns whether the struct identified by module_id and struct_id is the native U256 struct.
     pub fn is_u256(&self, struct_id: QualifiedId<StructId>) -> bool {
         let struct_env = self.env.get_struct(struct_id);
         attributes::is_evm_arith_module(&struct_env.module_env) && struct_env.is_native()
     }
 
     /// Check whether ty is a static type in the sense of serialization
-    pub fn abi_is_static_type(&self, ty: &Type) -> bool {
-        use move_model::ty::{PrimitiveType::*, Type::*};
-
-        let conjunction = |tys: &[Type]| {
+    pub fn abi_is_static_solidity_type(&self, ty: &SolidityType) -> bool {
```

Comment:
I suggest to move this and similar functions (basically everything which mostly works on `SolidityType`) into module solidity_type:

```
impl SolidityType {
  pub fn is_static(&self, ctx: &Context)
  pub fn size(&self) 
  ...
```

I think this will make the code better readable.

Technically, we have *not* done the same for `model::type::Type` because we cannot add impls to a type from another crate. But inside a crate we should maximize this for better readability.